### DISCUSSION TITLE
When will WebAssembly be supported?

### DISCUSSION THREAD

**Discussion by tumble29** [simolus3/drift#3000] [General]
I'm building my application for WebAssembly. I'm not familiar with WebAssembly, so initially I chose Drift to build a database for my web app thinking it has WASM support (because of the .wasm file). The app works fine on web but when building for WebAssembly `flutter build web --wasm` I got the error `The unavailable library 'dart:html' is imported through these packages:` (sqlite3). 

Browsing through issues I saw [your comment](https://github.com/simolus3/drift/issues/2745#issuecomment-1830742178) that WebAssembly support is planned in the future. Any news on when WebAssembly will be supported?

Thank you!

**Comment by simolus3** [comment]
The current state is that I have a [work-in-progress branch](https://github.com/simolus3/sqlite3.dart/tree/web) on the sqlite3 repository to no longer import `dart:html` or other legacy JS interop libraries. I hope to merge and release this soon, but that doesn't necessarily mean that compiling to WebAssembly will work.

Without substantially more work on Dart SDK tooling, WebAssembly support for `package:sqlite3` will always involve a JavaScript bridge between your Dart WASM app and the sqlite3 WASM module - so full and stable WebAssembly support will take a while. But once the migration to newer interop mechanisms is complete, I'll take a look at adding support for Dart apps compiled to WebAssembly.

**Reply by simolus3** [reply]
Version 2.4.3 of `package:sqlite3` supports being compiled to WebAssembly, although this is [very experimental for now](https://github.com/simolus3/sqlite3.dart/issues/230) and has severe restrictions. You can follow that issue for updates.

**Reply by tumble29** [reply]
Thank you for your work. I'm looking forward to integrating your amazing package into our app!
